packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -259,19 +470,36 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
   bool shouldRelayout(_ScaffoldLayout oldDelegate) {
     return oldDelegate.statusBarHeight != statusBarHeight
         || oldDelegate.bottomViewInset != bottomViewInset
-        || oldDelegate.endPadding != endPadding
-        || oldDelegate.textDirection != textDirection;
+        || oldDelegate.horizontalPadding != horizontalPadding
+        || oldDelegate.textDirection != textDirection
+        || !(fabMoveAnimation.isCompleted || fabMoveAnimation.isDismissed)
```

Comment:
this is sketchy

Reply:
Could you elaborate on the code smell you're seeing here?

The goal of this line is to have the layout continue when the animation is running.  Is there a better way to express this, or do you have a concern about this combination of a layout delegate and an animation?

Reply:
This method only gets called when a new object is created.
By definition, a delegate should always give the same result given the same constructor arguments.
What you're doing here "cheats" this; you recreate an identical object, but because it happens to be new, we call this method and then you get to say "oh by the way, even though we're identical, let's recompute, because we might have changed our mind".

The usual solutions for this would be to either:
 - not pass any mutable state into the delegate (e.g. pass in `Offset`s, not `Animation<Offest>`s), or
- make the delegate contract have a Listenable API, the way we did with the CustomPainter delegate.

Reply:
Done.  I've switched the animator to handling the offset directly rather than creating an animation to pass around.  This makes the getOffset code cleaner too, which is a nice plus.